### ISSUE TITLE
Extend configurability of initiative dialogs

### ISSUE THREAD

**Issue by ColmMassey** [SolidarityEconomyAssociation/sea-map#66]
The requirement to tailor panels and dialogs for different partners and use cases is growing.
What options are there to allow greater level of configurablity?
Use case 1 - Display homepage as a hyperlinked text of the URL rather than as a chain icon.
Use case 2 - If an X is a Qualifier of a Co-op then display this, but don't display its Organisational Type.

**Comment by ColmMassey** [comment]
Can you comment on how easy this is to do @dtmakm27 ?

**Comment by dtmakm27** [comment]
I guess we can use some reflection to do that and basically when we're building the map, import a standardized function with a specific signature from a specific file in the mini-website.

I will test out how fast i can make this happen, It's probably easier than it sounds

It's a good idea though, it's not good to include every customisation in one file with a hundred if-statements :) 

But on the other hand if it's not a simple/fast solution I think we should probably stick for now with the if-statements since we do not have a high demand for a feature like that (i think we have only a couple of use cases?). It's not that we cannot do it, but just the question of prioritization

**Comment by ColmMassey** [comment]
> But on the other hand if it's not a simple/fast solution I think we should probably stick for now with the if-statements since we do not have a high demand for a feature like that (i think we have only a couple of use cases?). It's not that we cannot do it, but just the question of prioritization

True. How many 'if' statements do we have at the moment?

**Comment by dtmakm27** [comment]
True. How many 'if' statements do we have at the moment?

one for dot coop for now


(two when we make the ica changes)

**Comment by dtmakm27** [comment]
I would have to include a popup.js file for all the map apps since there are some problems with loading a default template (most of them will be the same file) @ColmMassey

**Comment by ColmMassey** [comment]
Seems to work on the surface.

**Comment by wu-lee** [comment]
Just commenting retrospectively: As part of the work deploying the sea-map sites, in order to deploy the non-dotcoop sites I had to either make `popup.js` optional or copy the dotcoop `popup.js` into all the sea-site projects' `config` dirs.  I chose the harder case: to make it optional, which required a RequireJS plug-in to allow that without errors on building or page loading. I chose this in order to avoid a proliferation of cut-and-pasted code, which has been a problem before, and which makes the code hard to work with. 

I had to use the dotcoop pop-up as the default when nothing is supplied because that's all I had. However I believe that was the intention when Dean wrote, I think.
